### DISCUSSION TITLE
Can't get the real weights of the layers

### DISCUSSION THREAD

**Discussion by WD799** [tensorflow/compression#156] [Q&A] [answered]
when i use analysis_transform._layers[0].kernel and analysis_transform._layers[0].get_weights() the two functions, I get different weight. If i use analysis_transform._layers[0].kernel this function, i can get the real weights,but this kernel is read only that i can't modify them. And if i use get_weights() and set_weighs() , i get the wrong weights

**Comment by jonaballe** [comment, marked as answer]
Hi, this is because we use kernel DFT reparameterization by default, to improve training stability. So `.get_weights()` will return the DFTs of the kernels, and `.kernel` will return the kernel itself.

You can get the kernel using `layer.kernel`, and set it again using `layer.kernel_parameter = RDFTParameter(kernel, name='kernel')`.

Or you can turn off RDFT reparameterization by instantiating the layer with the keyword `kernel_parameter='variable'`.
